content/guides/weird_characters.adoc

```diff
@@ -0,0 +1,794 @@
+* http://clojure.github.io/clojure/clojure.core-api.html#clojure.core/\*out*[Clojure API Docs]
+
+== >!!, <!!, >! and <! - core.async channel macros
+
+These symbols are channel operations in `core.async` - a Clojure/ClojureScript
+library for channel based asynchronous programming (specifically http://en.wikipedia.org/wiki/Communicating_sequential_processes[CSP - Communicating Sequential Processes]).
+
+If you imagine, for the sake of argument, a channel is a bit like a queue that
+things can put stuff on and take stuff off, then these symbols support that
+simple API.
+
+* `>!!` and `<!!` are *blocking put* and *take* respectively
+* `>!` and `<!`are, simply *put* and *take*
+
+THe difference being the blocking version operate outside `go` blocks and block
+the tread they operate on.
+[source,clojure]
+----
+user=> (def my-channel (chan 10)) ; create a channel
+user=> (>!! my-channel "hello")   ; put stuff on the channel
+user=> (println (<!! my-channel)) ; take stuff off the channel
+hello
+----
+The non-blocking version sneed to be executed within a `go` block, otherwise
+they'll throw an exception.
+[source,clojure]
+----
+user=> (def c (chan))
+#'user/c
+user=> (>! c "nope")
+AssertionError Assert failed: >! used not in (go ...) block
+nil  clojure.core.async/>! (async.clj:123)
+----
+While the diffence between these is well outside the scope of this guide,
+fundamentally the `go` blocks operate and manage their own resources pausing
+*execution* of code without blocking threads. This makes asynchronously executed
+code appear to be synchronous, removing the pain of managing
+asynchronous code from the code base.
+
+* https://github.com/clojure/core.async/blob/master/examples/walkthrough.clj[core.async Code Walkthrough]
+* https://github.com/clojure/core.async/wiki[core.async Wiki]
+* <<xref/../core_async_go#,Go Block Best Practices>>
+
+== <symbol>? - Predicate Marker
+
+Putting `?` at the end of a symbol is a *naming convention* common across
+many languages that support special characters in their symbol names. It is
+used to indicate that the thing is a predicate, i.e. that it *poses a question*.
+For example, imagine using an API that delt with buffer manipulation:
+[source,clojure]
+----
+(def my-buffer (buffers/create-buffer [1 2 3]))
+(buffers/empty my-buffer)
+----
+At a glance, how would you know if the function `empty` in this case,
+* Returned `true` if the passed in buffer was empty, or,
+* Cleared the buffer
+While the author could have renamed `empty` to `is-empty`, the richness of
+symbol naming in Clojure allows us to express intent more symbolically.
+[source,clojure]
+----
+(def my-buffer (buffers/create-buffer [1 2 3]))
+(buffers/empty? my-buffer)
+false
+----
+This is simply a recommended *convension*, not a *requirement*.
+
+* https://github.com/bbatsov/clojure-style-guide#naming[Clojure Style Guide]
+
+== <symbol>! - Unsafe Operations
+
+The Clojure style guide has this to say
+
+[]
```

Comment:
Should link to https://github.com/bbatsov/clojure-style-guide#changing-state-fns-with-exclamation-mark

Maybe for googleability, droping the words `bang` and `exclamation mark` would also help